### PR TITLE
Bug: Fix Rich-Text Cross-Wiring in Comment Forms

### LINKED ISSUE #8478: Wrong text input gets selected while using toolbar on comments
While trying to add new comment  when any of the option is choosen from toolbar the changes are displayed in different text input.

ezgif com-video-to-gif (6)

### where to look

https://publiclab.org/questions/Cogreg/09-20-2020/how-do-i-test-home-made-fertilizer

### PR DESCRIPTION
**Pull request by noi5e**
Fixes #8478 

This PR is a duplicate of #9003, which is now closed (I did a `git rebase main` in the other branch, which was bundling changes from `main` into the PR. Couldn't figure out how to undo that).

From the other PR write-up:
> Interestingly, even before I merged unique button IDs, I saw that `e.target` was pretty accurate. That makes me think that these cross-wiring fixes maybe have less to do with unique IDs for every single element, and more to do with how the jQuery eventListeners are attached? Still not sure.

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #8775 for more context)

### CODE REVIEW

File: app/assets/javascripts/editor.js [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+    // assign dSelected
+    if (params.hasOwnProperty('dSelected')) {
+      $D.selected = params['dSelected'];
+    }
+    $E.initialize(params);
+    const action = e.currentTarget.dataset.action // 'bold', 'italic', etc.
+    $E[action](); // call the appropriate editor function
+  });
+});
+
```
Comment (noi5e) [inline app/assets/javascripts/editor.js:17]: I have doubts about if this is the right place to put this click eventHandler.

File: app/assets/javascripts/dragdrop.js [hunk 2]
```diff
@@ -32,20 +32,10 @@ jQuery(function() {
     $('#side-dropzone').removeClass('hover');
   });
   $('.dropzone').on('drop',function(e) {
-    // this 'drop' listener is also reused for pages with just one form, ie. /wiki/new
-    const closestCommentFormWrapper = e.target.closest('div.comment-form-wrapper'); // this returns null if there is no match
+    const params = getEditorParams(e.target); // defined in editorHelper.js
     e.preventDefault();
```
Comment (noi5e) [inline app/assets/javascripts/dragdrop.js:36]: Am I doing this right: calling on editorHelper.js? I browsed the Rails guide to JS assets, but am not sure if what I'm doing falls under 'best practices'. Would appreciate some pointers if you have any.

**Comment by noi5e** [comment]
@jywarren Had some questions about requiring JS files in Rails, and where to put a click eventHandler.

Other than that, this is ready to merge!

File: app/assets/javascripts/dragdrop.js (hunk 2 above)
Comment (sagarpreet-chadha) [reply app/assets/javascripts/dragdrop.js:36]: Yes that looks fine to me

**Review by sagarpreet-chadha — APPROVED** [review]
🎉

**Review by cesswairimu — APPROVED** [review]
Fantastic :tada:

**Pull request merged**